examples/todos/common/src/commonMain/kotlin/org/reduxkotlin/examples/todos/AppState.kt

```diff
@@ -0,0 +1,35 @@
+package org.reduxkotlin.examples.todos
+
+
+/**
+ * Entire state tree for the app.
+ */
+data class AppState(
+    val todos: List<Todo> = listOf(),
+    val visibilityFilter: VisibilityFilter = VisibilityFilter.SHOW_ALL
+) {
+    val visibleTodos: List<Todo>
+        get() = getVisibleTodos(visibilityFilter)
+
+    private fun getVisibleTodos(visibilityFilter: VisibilityFilter) = when (visibilityFilter) {
+        VisibilityFilter.SHOW_ALL -> todos
+        VisibilityFilter.SHOW_ACTIVE -> todos.filter { !it.completed }
+        VisibilityFilter.SHOW_COMPLETED -> todos.filter { it.completed }
+    }
+}
+
+/**
+ * A model of data in out state.  Where these types of models are located is up to you and your team
```

Comment:
Do you just mean view state here? data in/out state is a tad bit confusing

```suggestion
 * A model of data in/out state.  Where these types of models are located is up to you and your team
```

Reply:
Good catch.  I was saying "data in our state".  I've updated to say "A model used in the app state"